### PR TITLE
Fix `<FormDataConsumer>` usage with Field children

### LINKED ISSUE #8427: Fields are not reactive when used inside `SimpleFormIterator`
Related to #8359

None of the possible solutions work correctly when using a `Field` inside a `SimpleFormIterator`.
I put together a Stackblitz (https://stackblitz.com/edit/github-stqpcj?file=src%2Fposts%2FPostEdit.tsx) to show what I mean.

It shows tree components:
1- `Labeled` + `FunctionField`. => It works but with a warning for not using `getSource`
2- `Labeled` + `TextField` passing `source='url'` and `record={scopedFormData}`. => It works but with a warning for not using `getSource` 
3- `Labeled` + `TextField` with `source=getSource('url')`. => Shows the data but doesn't change when an input of the same source  changes (merged in #8359)

Reproduction:
1- Edit the Post of id=1 and click in Miscellaneous Tab.
2- In the Backlinks `ArrayInput`, type into the URL input to see the results

### PR DESCRIPTION
**Pull request by WiXSL**
Fixes #8427

### CODE REVIEW

**Review by slax57 — APPROVED** [review]
I approve the changes.

I would just suggest changing the PR title to something more explicit. For instance something like:
Fix `<SimpleFormIterator>` example on how to work with Fields, and remove confusing warning in `<FormDataConsumer>`

**Review by djhi — APPROVED** [review]

File: docs/SimpleFormIterator.md [hunk 1]
```diff
@@ -156,9 +156,9 @@ import { ArrayInput, SimpleFormIterator, DateInput, TextField, FormDataConsumer,
     <SimpleFormIterator disableRemove>
         <DateInput source="date" />
         <FormDataConsumer>
-            {({ getSource }) => (
+            {({ scopedFormData }) => (
                 <Labeled label="Url">
-                    <TextField source={getSource('url')} />
+                    <TextField source="url" record={scopedFormData} />
```
Comment (djhi) [inline docs/SimpleFormIterator.md:161]: Are you sure about the removal of `getSource` ?

Comment (slax57) [reply docs/SimpleFormIterator.md:161]: Yes, this solution was tested in the context of this PR: https://github.com/marmelab/react-admin/pull/8359 (have a look at the comments)

Comment (djhi) [reply docs/SimpleFormIterator.md:161]: Does it mean we don't need `getSource` anymore?

**Review by fzaninotto — APPROVED** [review]

File: docs/SimpleFormIterator.md (hunk 1 above)
Comment (fzaninotto) [reply docs/SimpleFormIterator.md:161]: we still need it for inputs

**Pull request merged**